src/libraries/System.Private.Xml/src/System/Xml/Schema/XsdDuration.cs

```diff
@@ -410,7 +426,8 @@ internal string ToString(DurationType durationType)
                             }
 
                             vsb.EnsureCapacity(zeroIdx + 1);
-                            vsb.Append(tmpSpan.Slice(0, zeroIdx - len + 1));
+                            var nanoSpanLength = zeroIdx - len + 1;
+                            tmpSpan[..nanoSpanLength].TryCopyTo(vsb.AppendSpan(nanoSpanLength));
```

Comment:
should we check result at least in debug mode? btw. I personally find original syntax way more readable than this - any particular reason for change?

Reply:
When you initialize ValueStringBuilder with Span<char> argument you receive following error :
![image](https://user-images.githubusercontent.com/12857389/197526875-01d293ed-8614-4b5f-852a-ef56329288b5.png)

Code before works only if I define vsb like this
![image](https://user-images.githubusercontent.com/12857389/197527094-a1973a0d-4b1d-4bec-9a18-5607e82d9705.png)

not like :
![image](https://user-images.githubusercontent.com/12857389/197527175-8cd3aaa0-5ab0-4bf0-96c0-3ea26972798c.png)